core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Range.hpp

```diff
@@ -82,131 +125,299 @@ class Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType,
+        reducer.join(&temp, &gang_red_temp(idx));
+      }
+      KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(thread_id) = temp;
+    }
+#pragma acc loop seq
+    for (tStep = (chunk_size >> 1); tStep > 0; tStep >>= 1) {
+#pragma acc loop vector
+      for (IndexType thread_id = 0; thread_id < chunk_size; ++thread_id) {
+        if (thread_id < tStep) {
+          reducer.join(
+              &KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(thread_id),
+              &KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(thread_id + tStep));
+        }
+        IndexType checkOdd = tSize & 1;
+        if (checkOdd == 1) {
+          if (thread_id == 0) {
+            reducer.join(
+                &KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(thread_id),
+                &KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(tSize - 1));
+          }
+        }
+      }
+      tSize = tStep;
+    }
+    reducer.final(&KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(0));
+    if (m_result_on_device) {
+      *m_result_ptr = KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(0);
+    } else {
+      m_result_view() = KOKKOS_IMPL_OPENACC_VECTOR_RED_TEMP_ACCESS2(0);
+    }
+  }
+
+  if (!m_result_on_device) {
+    Kokkos::Impl::DeepCopy<HostSpace, Kokkos::Experimental::OpenACCSpace,
+                           Kokkos::Experimental::OpenACC>(
+        policy.space(), m_result_ptr, m_result_view.data(), sizeof(ValueType));
+  }
+
+  acc_wait(async_arg);
+#pragma acc exit data delete (functor, reducer, gang_red_temp)
+}
+}  // namespace Kokkos::Experimental::Impl
+
+template <class Functor, class Reducer, class... Traits>
+struct Kokkos::Experimental::Impl::OpenACCParallelReduceHelper<
+    Functor, Reducer, Kokkos::RangePolicy<Traits...>, true> {
+  using Policy       = RangePolicy<Traits...>;
+  using ScheduleType = Kokkos::Experimental::Impl::OpenACCScheduleType<Policy>;
+  using Pointer      = Kokkos::Impl::FunctorAnalysis<
+      Kokkos::Impl::FunctorPatternInterface::REDUCE, Policy,
+      typename Functor::functor_type,
+      typename Reducer::value_type>::pointer_type;
+
+  OpenACCParallelReduceHelper(Functor const& functor, Reducer const& reducer,
+                              Policy const& policy, Pointer m_result_ptr,
+                              bool m_result_on_device) {
+    auto const begin = policy.begin();
+    auto const end   = policy.end();
+
+    if (end <= begin) {
+      return;
+    }
+
+    OpenACCParallelReduceCustom(ScheduleType(), policy, functor, reducer,
+                                m_result_ptr, m_result_on_device);
+  }
+};
+
+#define KOKKOS_IMPL_OPENACC_PARALLEL_REDUCE_DISPATCH_SCHEDULE(REDUCER,         \
+                                                              OPERATOR)        \
+  namespace Kokkos::Experimental::Impl {                                       \
+  template <class Policy, class Functor, class Reducer, class Pointer>         \
+  void OpenACCParallelReduce##REDUCER(Schedule<Static>, Policy const& apolicy, \
```

Comment:
Why are we changing the signature and now passing the policy as argument?
It looked intentional to unpack at call site and pass `chunk_size`, `begin`, `end`, and `async_arg`.
It increases the diff w/o obvious benefit.  What am I missing?

Reply:
It is mainly to make both `OpenACCParallelReduceCustom()`  in line 298 and `OpenACCParallelReduce##REDUCER()` in line 419 have the same function signature.
    ==> `OpenACCParallelReduceCustom()` and `OpenACCParallelReduce##REDUCER()` use different policy member data (e.g., `OpenACCParallelReduceCustom()` uses `policy.space()` but `OpenACCParallelReduce##REDUCER()` doe not use it.)